internal/service/dfp/api_dfp_resource.go

```diff
@@ -0,0 +1,178 @@
+package dfp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	bloxoneclient "github.com/infobloxopen/bloxone-go-client/client"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/utils"
+	"net/http"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &DfpResource{}
+var _ resource.ResourceWithImportState = &DfpResource{}
+
+func NewDfpResource() resource.Resource {
+	return &DfpResource{}
+}
+
+// DfpResource defines the resource implementation.
+type DfpResource struct {
+	client *bloxoneclient.APIClient
+}
+
+func (r *DfpResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "dfp_service"
+}
+
+func (r *DfpResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Manages a DNS Forwarding Proxy Service.",
+		Attributes:          DfpResourceSchemaAttributes,
+	}
+}
+
+func (r *DfpResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*bloxoneclient.APIClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *bloxoneclient.APIClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *DfpResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data DfpModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, _, err := r.client.DNSForwardingProxyAPI.
+		InfraServicesAPI.
+		CreateOrUpdateDfpService(ctx, utils.ExtractResourceId(data.ServiceId.ValueString())).
```

Comment:
`ExtractResourceId` should be handled in go-client like we do for all other resources.

Reply:
Fixed